apps/evm_family/evm_api.h

```diff
@@ -0,0 +1,113 @@
+/**
+ * @file    evm_api.h
+ * @author  Cypherock X1 Team
+ * @copyright Copyright (c) 2023 HODL TECH PTE LTD
+ * <br/> You may obtain a copy of license at <a href="https://mitcc.org/"
+ * target=_blank>https://mitcc.org/</a>
+ */
+#ifndef EVM_API_H
+#define EVM_API_H
+
+/*****************************************************************************
+ * INCLUDES
+ *****************************************************************************/
+
+#include <evm/core.pb.h>
+#include <stddef.h>
+#include <stdint.h>
+
+/*****************************************************************************
+ * MACROS AND DEFINES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * TYPEDEFS
+ *****************************************************************************/
+
+/*****************************************************************************
+ * EXPORTED VARIABLES
+ *****************************************************************************/
+
+/*****************************************************************************
+ * GLOBAL FUNCTION PROTOTYPES
+ *****************************************************************************/
+
+/**
+ * @brief API to decode query from host with `EVM_QUERY_FIELDS`
+ *
+ * @param[in] data: PB encoded bytestream received from host
+ * @param[in] data_size: size of pb encoded bytestream
+ * @param[out] query_out: @ref evm_query_t obj to copy the decoded result to
+ * @return bool True if decoding was successful, else false
+ */
+bool decode_evm_query(const uint8_t *data,
+                      uint16_t data_size,
+                      evm_query_t *query_out);
+
+/**
+ * @brief Encodes the result with `EVM_RESULT_FIELDS` to byte-stream
+ *
+ * @param[in] result: object of populated @ref evm_result_t to be encoded
+ * @param[out] buffer: buffer to fill byte-stream into
+ * @param[in] max_buffer_len: Max length allowed for writing bytestream to
+ * buffer
+ * @param[out] bytes_written_out: bytes written to bytestream
+ * @return bool True if decoding was successful, else false
+ */
+bool encode_evm_result(evm_result_t *result,
```

Comment:
let's make the function param `const`

better to have it as const at all the similar places (`encode_btc_result`, `encode_manager_result`) in this PR itself

```suggestion
bool encode_evm_result(const evm_result_t *result,
```